backend/server.py

```
@@ -1086,8 +1086,116 @@ async def admin_update_affiliate(
+    )
+    session.add(subscriber)
+    await session.commit()
+
+    return {"success": True, "id": subscriber.id, "email": subscriber.email}
+
+
+@api_router.post("/auth/newsletter/subscribe")
+async def auth_newsletter_subscribe(
+    payload: NewsletterSubscribeRequest,
+    request: Request,
+    session: AsyncSession = Depends(get_db)
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)